### PR TITLE
Rework Python binding

### PR DESCRIPTION
**Pull request by zxzxwu**
- Add pytests
- Rename arguments to make them more clear
- Add type annotations
- Black reformat

### CODE REVIEW

**Comment by zxzxwu** [comment]
Hi @asoulier, could you help review this PR? Thanks.

File: python/tools/decoder.py [hunk 1]
```diff
@@ -80,7 +80,7 @@
 for i in range(0, encoded_length, frame_length):
 
     lc3_frame_size = struct.unpack('=H', f_lc3.read(2))[0]
-    pcm = dec.decode(f_lc3.read(lc3_frame_size), bitdepth=bitdepth)
+    pcm = dec.decode(f_lc3.read(lc3_frame_size), bit_depth=bitdepth)
```
Comment (asoulier) [inline python/tools/decoder.py:83]: I think the argument name can be changed to `bit_depth` too.

File: python/lc3.py [hunk 2]
```diff
@@ -13,96 +13,124 @@
...
-
-            lib.lc3_hr_delay_samples = \
-                lambda hrmode, dt_us, sr_hz: \
-                lib.lc3_delay_samples(dt_us, sr_hz)
+                raise InitializationError("High-Resolution interface not available")
+
+            lc3_hr_frame_samples = lambda hrmode, dt_us, sr_hz: lib.lc3_frame_samples(
+                dt_us, sr_hz
+            )
+            lc3_hr_frame_block_bytes = (
```
[the remark this answers is not in the document]
Comment (zxzxwu) [reply python/lc3.py:108]: It's formatted by Black automatically (though I didn't specify a config). When a line is too long, Black will saperate it as a new line which requires a new parenthesis in Python, but some lines are not so long.

File: python/lc3.py [hunk 3]
```diff
@@ -13,96 +13,124 @@
...
+
+        self.hrmode = hrmode
+        self.frame_duration_us = frame_duration_us
+        self.sample_rate_hz = sample_rate_hz
+        self.pcm_sample_rate_hz = pcm_sample_rate_hz or self.sample_rate_hz
+        self.num_channels = num_channels
+
+        if self.frame_duration_us not in [2500, 5000, 7500, 10000]:
+            raise InvalidArgumentError(
+                "Invalid frame duration: %.1f ms" % self.frame_duration_us
```
[the remark this answers is not in the document]
Comment (zxzxwu) [reply python/lc3.py]: Done

File: python/tools/decoder.py (hunk 1 above)
Comment (zxzxwu) [reply python/tools/decoder.py]: Done

File: python/lc3.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (asoulier) [reply python/lc3.py:108]: Got it, thanks.

**Pull request merged**